R/irace_helpers.R

```diff
@@ -0,0 +1,81 @@
+paradox_to_irace = function(ps) {
+
+  assertClass(ps, "ParamSet")
+  # what about ParamUty = vector numeric/real
+  class_lookup = data.table(
+    paradox = c("ParamLgl", "ParamInt", "ParamDbl", "ParamFct"),
+    irace = c("c", "i", "r", "c"), stringsAsFactors = FALSE)
+
+  type = unlist(subset(merge(data.table(paradox = ps$class), class_lookup, sort = FALSE),
+    select = "irace"))
+  range = get_irace_range(ps)
+  if (ps$has_deps) {
+    condition = get_irace_condition(ps)
+  } else {
+    condition = NULL
+  }
+
+  par_tab = paste(ps$ids(), '""', type, range, condition$cond, collapse = "\n")
+
+  return(irace::readParameters(text = par_tab))
+}
+get_irace_range = function(ps) {
+  rng = data.table(lower = ps$lower, upper = ps$upper, lvl = ps$levels)
+
+  apply(rng, 1, function(x) {
+    if (is.na(x[[1]])) {
+      return(paste0("(", paste0(x[[3]], collapse = ","), ")"))
+    } else {
+      return(paste0("(", x[[1]], ",", x[[2]], ")"))
+    }
+  })
+}
+get_irace_condition = function(ps) {
+  cond = rbindlist(apply(ps$deps, 1, function(x) {
+    on = x[[2]]
+    cond = x[[3]]$rhs
+    if (is.character(cond)) {
+      cond = paste0("'", cond, "'")
+    }
+    if (x[[3]]$type == "equal") {
+      condition = paste("|", x[[2]], "==", cond)
+    } else {
+      condition = paste("|", x[[2]], "%in%", paste0("c(", paste0(cond, collapse = ","), ")"))
+    }
+    data.table(id = x[[1]], cond = condition)
+  }))
+
+  # coercion back and forth from frame/table is due to data.frame sorting even when sort = FALSE
+  tab = data.frame(merge(data.table(id = ps$ids()), cond, by = "id", all.x = TRUE, sort = FALSE))
+  tab[is.na(tab)] = ""
+
+  return(tab)
+}
+make_scenario = function(inst) {
+  list(
+    targetRunner = targetRunner,
+    logFile = tempfile(),
+    instances = list(inst),
+    debugLevel = 0,
+    maxExperiments = if (class(inst$terminator)[1] == "TerminatorEvals")
+      inst$terminator$param_set$values$n_evals else 0,
+    maxTime = if (class(inst$terminator)[1] == "TerminatorRunTime")
+      inst$terminator$param_set$values$secs else 0
+  )
+}
+
+targetRunner = function(experiment, scenario) { # nolint
+  t0 = Sys.time()
+  # fix logicals
+  config = as.data.table(lapply(experiment$configuration, function(x) {
+    if (x %in% c("TRUE", "FALSE")) {
+      return(as.logical(x))
+    } else {
+      return(x)
+    }
+  }))
+  cost = scenario$instances[[1]]$eval_batch(config)[[1]]
```

Comment:
> The objective of irace is to minimize the cost value returned by the target algorithm. If you wish to maximize, you can multiply the cost by -1 before returning it to irace.

(irace documentation)

Do you minimize somewhere else? If not you can use `scenario$instances[[1]]$objective_function(config)` which always minimizes.

Reply:
changed to `objective_function`

Reply:
Sorry but we have to use `$eval_batch()` again. I allows us to pass extra information like `id.instance` to the archive.

Reply:
Okay, change whatever needs changing!